ProjectCoimbra.UWP/Project.Coimbra/Pages/LaneSettingsPage.xaml.cs

```diff
@@ -220,7 +220,7 @@ private void OptimizeKeyboard_Click(object sender, RoutedEventArgs e)
                 lane.PropertyChanged += this.Lane_PropertyChanged;
                 this.Lanes.Children.Add(lane);
 
-                foreach (var note in this.AvailableNotes.Where(note => note.Value.StartsWith(notes[currentAdd], StringComparison.OrdinalIgnoreCase)))
+                foreach (var note in this.AvailableNotes.Where(note => note.Value.StartsWith(notes[currentAdd], StringComparison.OrdinalIgnoreCase)).ToList())
```

Comment:
I think `ToList()` might actually degrade performance here as the list is just iterated through once, assuming that the query to construct the iterator is just calculated once (which it should be). An `IEnumerable` is efficient if iterated through just a single time -- in this case, using `ToList()` is worse as it iterates once to create the list and then you need to go through each element again.